tools/scripts/release/auto_build.py

```diff
@@ -113,15 +111,18 @@ def setup_cmake_dir(self, config):
             # TODO - check for valid version string
             cmd += [f"-DVERSION={self.version}"]
 
-        # If we built a bootloader, let's include it in the main app
-        # TODO - don't create unified image when encryption is enabled,
-        # since private key will be present on it (and unecrypted)
-        if (
-            config["name"] != "bootloader"
-            and "bootloader" in self._configs
-            and "elf_fullpath" in self._configs["bootloader"]
-        ):
-            cmd += [f'-DBOOTLOADER_PATH={self._configs["bootloader"]["elf_fullpath"]}']
+        # If we built a bootloader, let's create a unified image, unless encryption
+        # is enabled, since the private key would be present in it and unecrypted.
+        if not config["name"].startswith("bootloader") and not config["encrypt"]:
+            bootloader_cfg_key = (
+                "bootloader_signing_required"
+                if config["sign"]
+                else "bootloader_development"
+            )
+            if bootloader_cfg_key in self._configs:
+                bootloader_cfg = self._configs[bootloader_cfg_key]
+                if "elf_fullpath" in bootloader_cfg:
+                    cmd += [f'-DBOOTLOADER_PATH={bootloader_cfg["elf_fullpath"]}']
```

Comment:
This was the main slightly tricky thing — include the correct bootloader in the unified image based on whether we're signing the app. Also took care of the TODO comment.